src/IRCNetwork.cpp

```diff
@@ -1151,7 +1151,11 @@ CServer* CIRCNetwork::GetNextServer() {
 		m_uServerIdx = 0;
```

Comment:
Since this only moves the pointer when it already points past the end of the list, it's technically not advancing the index, just setting it to where it "should be".  Unless I'm missing something, the next call to GetNextServer with `bAdvance == true` would have just set the index to zero anyways, and this way we can return what that next server would have been without having to do a special check.